frontend/src/components/main-view/details/Details.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react'
+import {useAppDispatch, useAppSelector} from "../../../app/hooks";
+import {
+    hideDetails,
+    resetDetails,
+    resetGoBack,
+    selectGoBack,
+    selectShowDetails,
+    setDetailData
+} from "../../../slicer/detailsSlice";
+import { selectCurrentSupplier } from '../../../slicer/supplierSlice';
+import {parseSupplierToThumbnailData} from "../../../interfaces/IThumbnailData";
+//component imports
+import {
+    Container,
+    Dialog,
+    DialogContent,
+     Fab,
+    useMediaQuery,
+    useTheme
+} from "@mui/material";
+import ClearIcon from '@mui/icons-material/Clear'
+import ArrowBackIcon from '@mui/icons-material/ArrowBack'
+import DetailsCard from "./details-card/DetailsCard";
+
+
+//interface imports
+
+type Props = {};
+
+
+function Details(props: Props) {
+    const dispatch = useAppDispatch();
+    const showDetails = useAppSelector(selectShowDetails);
+    const supplier = useAppSelector(selectCurrentSupplier);
+    const goBack = useAppSelector(selectGoBack);
+    const descriptionElementRef = React.useRef<HTMLDivElement>(null);
+    const theme = useTheme();
+    const fullScreen = useMediaQuery(theme.breakpoints.down('sm'));
+    const alignItems = fullScreen ? "center" : "";
+    React.useEffect(() => {
```

Comment:
Is there a reason why you write React.useEffect here instead of importing useEffect?